internal/remote/remote_catalog.go

```diff
@@ -102,9 +124,96 @@ func (c *RemoteCatalog) Delete(
 ) error {
 	skrContext, err := c.skrContextFactory.Get(kyma)
 	if err != nil {
-		return fmt.Errorf("failed to get SkrContext for deleting RemoteCatalog: %w", err)
+		return fmt.Errorf("failed to get SKR context: %w", err)
 	}
 
 	moduleTemplates := c.moduleTemplateSyncAPIFactoryFn(c.kcpClient, skrContext.Client, &c.settings)
 	return moduleTemplates.DeleteAllManaged(ctx)
 }
+
+// GetModuleReleaseMetasToSync returns a list of ModuleReleaseMetas that should be synced to the SKR.
+// A ModuleReleaseMeta that is Beta or Internal is synced only if the Kyma is also Beta or Internal.
+func (c *RemoteCatalog) GetModuleReleaseMetasToSync(
+	ctx context.Context,
+	moduleReleaseMetas *[]v1beta2.ModuleReleaseMeta,
+	kyma *v1beta2.Kyma,
+) error {
+	moduleReleaseMetaList := &v1beta2.ModuleReleaseMetaList{}
+	if err := c.kcpClient.List(ctx, moduleReleaseMetaList); err != nil {
+		return fmt.Errorf("failed to list ModuleReleaseMetas: %w", err)
+	}
+
+	for _, moduleReleaseMeta := range moduleReleaseMetaList.Items {
+		if moduleReleaseMeta.IsBeta() && !kyma.IsBeta() {
+			continue
+		}
+		if moduleReleaseMeta.IsInternal() && !kyma.IsInternal() {
+			continue
+		}
+		*moduleReleaseMetas = append(*moduleReleaseMetas, moduleReleaseMeta)
+	}
+
+	return nil
+}
+
+// GetModuleTemplatesToSync returns a list of ModuleTemplates that should be synced to the SKR.
+// A ModuleTemplate is synced if it is not mandatory and does not have sync disabled. In addition,
+// it must be referenced by a ModuleReleaseMeta that is synced.
+func (c *RemoteCatalog) GetModuleTemplatesToSync(
+	ctx context.Context,
+	moduleTemplates *[]v1beta2.ModuleTemplate,
+	moduleReleaseMetas *[]v1beta2.ModuleReleaseMeta,
```

Comment:
'moduleReleaseMetas' doesn't have to be a slice pointer here. It makes the code less readable and implies that the function requires a pointer - maybe it's re-allocating the target object? It's not the case here.